source/administration/config-settings.rst

```diff
@@ -3108,6 +3131,30 @@ Prefix on the Elasticsearch index name. Enables the use of Mattermost Elasticsea
 | This feature's ``config.json`` setting is ``"IndexPrefix": ""`` with string input      |
 +----------------------------------------------------------------------------------------+
 
+Live Indexing Batch Size
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+XXXXXX
+
++--------------------------------------------------------------------------------------------------------+
+| This feature's ``config.json`` setting is ``"LiveIndexingBatchSize": 1"`` with whole number input      |
++--------------------------------------------------------------------------------------------------------+
+
+Request Timeout
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Timeout in seconds for Elasticseaerch calls.
+
++-------------------------------------------------------------------------------------------------------+
+| This feature's ``config.json`` setting is ``"RequestTimeoutSeconds": 30`` with whole number input     |
++-------------------------------------------------------------------------------------------------------+
+
+Bulk Indexing Time Window
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+XXXXXX
```

Comment:
@grundleborg

Reply:
This value sets the maximum length of the time window for a batch of posts being indexed by the Bulk Indexer. It works in conjunction with the hard coded limit of a number of posts in a batch, with the smaller of the two being the ultimate limiting factor for batch size.

It serves as a performance optimisation for very large installs (as a rough guideline, >10 million posts in the database). On smaller installs, it should not be changed from its default value.